tcms/testruns/migrations/0012_test_execusion_add_fields.py

```diff
@@ -0,0 +1,43 @@
+# Generated by Django 3.1.6 on 2021-02-17 04:46
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("testruns", "0011_add_planning_fields"),
+    ]
+
+    operations = [
+        migrations.RenameField(
+            model_name="historicaltestexecution",
+            old_name="close_date",
+            new_name="stop_date",
+        ),
+        migrations.RenameField(
+            model_name="testexecution",
+            old_name="close_date",
+            new_name="stop_date",
+        ),
+        migrations.AddField(
+            model_name="historicaltestexecution",
+            name="start_date",
+            field=models.DateTimeField(blank=True, db_index=True, null=True),
+        ),
+        migrations.AddField(
+            model_name="historicaltestexecution",
+            name="stop_date",
+            field=models.DateTimeField(blank=True, db_index=True, null=True),
+        ),
```

Comment:
This is causing 
`django.db.utils.ProgrammingError: column "stop_date" of relation "testruns_historicaltestexecution" already exists` in test-migrations-rollback. Remove it.

Reply:
I checked **test-migrations-rollback**  but I cant find anywhere where it `references the stop_date of relation "testruns_historicaltestexecution" already exists ` so that I can remove it. 

This is what that it shows: 
` test-migrations-rollback:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        python-version: [3.7]

    steps:
    - uses: actions/checkout@v2
    - name: Set up Python ${{ matrix.python-version }}
      uses: actions/setup-python@v1
      with:
        python-version: ${{ matrix.python-version }}

    - name: Create database
      run: |
        docker-compose -f docker-compose.postgres pull db
        docker-compose -f docker-compose.postgres run -d -p 5432:5432 --name kiwi_db db

    - name: migrations rollback
      run: |
        pip install -r requirements/devel.txt
        pip install -r requirements/postgres.txt
        pushd tcms/ && npm install && popd

        docker exec -i kiwi_db psql -c "ALTER USER kiwi CREATEDB;"

        export LANG=en-us
        # Postgres b/c it has best support for schema altering operations
        export DJANGO_SETTINGS_MODULE=tcms.settings.test.postgresql
        ./tests/migrations-rollback`

Do you know of any other ways to check where this duplicated column is created? I also searched where else the stop_date is being referenced but I'm not sure which specific code creates that extra column.

Reply:
Migration operations are executed in the order they are defined in the migration file. You are renaming an older field to `stop_date` 15-20 lines above and as a last operation you are trying to add a new field with the same name which obviously doesn't work. 

This is a left-over from your 1st and 2nd commits. Checkout your git log/individual commits for history.